lib/oneview-sdk/rest.rb

```diff
@@ -121,6 +124,84 @@ def rest_delete(path, options = {}, api_ver = @api_version)
       rest_api(:delete, path, options, api_ver)
     end
 
+    # Uploads a file to a specific uri
+    # @param [String] file_path
+    # @param [String] uri The uri starting with "/"
+    # @param [Hash] body_params The params to append to body of http request. Default is {}.
+    # @option body_params [String] 'name' The name to show (when resource accepts a name)
+    # @param [Integer] timeout The number of seconds to wait for completing the request. Default is 300.
+    # @return [Hash] The parsed JSON body of response
+    def upload_file(file_path, uri, body_params = {}, timeout = READ_TIMEOUT)
+      raise NotFound, "ERROR: File '#{file_path}' not found!" unless File.file?(file_path)
+      options = {
+        'Content-Type' => 'multipart/form-data',
+        'X-Api-Version' => @api_version.to_s,
+        'auth' => @token
+      }
+      final_uri = URI.parse(URI.escape("#{@url}#{uri}"))
+
+      File.open(file_path) do |file|
+        name_to_show = body_params.delete('name') || body_params.delete(:name)
+        if name_to_show && !name_to_show.empty?
+          name_to_show.chomp!(File.extname(name_to_show)) unless File.extname(name_to_show).empty?
+          name_to_show += File.extname(file_path)
```

Comment:
But you should not be able to change the extension of file, right?
If you change the extension, file will look as having an another extension and oneview will throw error.